pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -759,6 +736,25 @@ func (r *RootSyncReconciler) validateSpec(ctx context.Context, rs *v1beta1.RootS
 	}
 }
 
+// checkValuesFileSourcesRefs validates that the ConfigMaps specified in the RSync ValuesFileSources exist and have the
+// specified data key.
+func (r *RootSyncReconciler) checkValuesFileSourcesRefs(ctx context.Context, cl client.Client, rs *v1beta1.RootSync) status.Error {
```

Comment:
also, when it's on the struct, we dont need to pass in the client.

Reply:
I moved as much of this check as I could into the validation function to avoid the repetition, and am no longer passing in the client as a separate argument